pytorch_lightning/plugins/training_type/ddp.py

```diff
@@ -155,19 +156,24 @@ def _call_children_scripts(self):
         os.environ["NODE_RANK"] = str(self.cluster_environment.node_rank())
         os.environ["LOCAL_RANK"] = str(self.cluster_environment.local_rank())
 
-        # when user is using hydra find the absolute path
-        path_lib = os.path.abspath if not _HYDRA_AVAILABLE else to_absolute_path
-
-        # pull out the commands used to run the script and resolve the abs file path
-        command = sys.argv
-        try:
-            full_path = path_lib(command[0])
-        except Exception:
-            full_path = os.path.abspath(command[0])
-
-        command[0] = full_path
-        # use the same python interpreter and actually running
-        command = [sys.executable] + command
+        # Check if the current calling command looked like `python a/b/c.py` or `python -m a.b.c`
+        # See https://docs.python.org/3/reference/import.html#main-spec
+        if __main__.__spec__ is None:  # Script called as `python a/b/c.py`
```

Comment:
can codecov follow this code path? we can put a `# pragma: no-cover`

Reply:
This path should be tested in the 3 modified tests with `as_module=False`. I don't know about the internals of how codecov works, so if it recognizes that this is in fact tested.

Should I just add that pragma?

Reply:
I think it is because we launch our tests with `coverage run blabla`, but these tests you saw then calls a regular python script from there, `python train.py ` etc so coverge doesn't run from there and we will run the code under
`if __main__.__spec__ is None:` but coverage is not captured. So I suggest:

`if __main__.__spec__ is None:  # pragma: no-cover`